### ISSUE TITLE
Submit stations listed in last update that are missing from radio-browser latest update.

### ISSUE THREAD

**Issue by JazzfanRS** [JazzfanRS/Radiosure-station-database#5]
This is a clerical issue that is a good fit for myself as I am not a programmer. 

**stations-2022-03-10.rsd** has few stations in common with the latest radio-browser update. 
RadioSure is gone, but the last update shouldn't be ignored and the missing stations should be submitted to radio-browser.

This is a good fit for myself as I am not a programmer. This is purely data entry and word processing that is best left to one or two people. I am extracting a list of the missing stations, and will post a link to the list. (I will be referring to this list and editing it as stations are submitted)

**Comment by JazzfanRS** [comment]
Anyone better versed at regex to use in np++ (preferred) or know a script for spreadsheets to sort a column by character count? I've just started learning FUNCTION and can't get LEN to work on 30000 rows. Much thanks.

**Comment by GHbasicuser** [comment]
If you use a spreadsheet, the easiest way is perhaps to do this by adding a column ("hidden" if necessary) in which you put the formula for each of the rows : **=NBCAR(C1)** or **=LEN(C1)**   ("C" = name of the column)

**Comment by JazzfanRS** [comment]
The problem I find is after placing the function the tutorial says to grab it and drag it to include all the rows to be sorted.  I will give this a try.

**Comment by JazzfanRS** [comment]
Okay, thanks for the hint. I had just joined [https://www.reddit.com/r/googlesheets/](https://www.reddit.com/r/googlesheets/ ) while I tried to figure it out after I asked you. I got another solution. 

`=ARRAYFORMULA(LEN(A:A))

Or if you just want a count of everything summed up, wrap that equation in SUM()`

**Comment by GHbasicuser** [comment]
With Calc (Libreoffice) there is also a similar way 😉, using the function wizard, check the "**Matrix**" box (structure area) and apply this formula :

> (=NBCAR(A:A))

 ia : `=NBCAR(A1:A31561)`

**Comment by JazzfanRS** [comment]
This quite helpful. I wonder if I should stick to 'management' and just come up with tasks for you two :)

**Comment by GHbasicuser** [comment]
I haven't looked at the content of the 2 "RSD" files but I really doubt that there are so few radios in common. 🤔 That's a lot of work in sight for 2-3 people ! 😲 (it will be without me 😜)

I believe that the least tedious solution for you would be to invite RadioSure users to report radios that they can no longer find. 🧐 And then that volunteers try to find these radios to add them to Radio-Browser.

This could be done through a collaborative spreadsheet board (online, free and without traps) like this one 👍 :  https://framacalc.org/abc/fr/

And if they know the streaming link and other information relating to a radio station, they can add it themselves to the Radio-Browser database, this will help everyone 😅 : https://www.radio-browser.info/add

**Comment by JazzfanRS** [comment]
I was wrong. The number in common is a lot less then 200, but anyway I updated my [Reddit post](https://www.reddit.com/r/RADIOSURE/comments/uh12pq/bookmark_this_post_date_below_will_change_with/?utm_source=share&utm_medium=web2x&context=3) that will have the link to @francois-neosurf's server, (see what I did there?) but the subtle invite is buried in TL;DR right now. There are only 8 members, and any number readers that don't join or comment. 
I think I was waiting for them to be able to download a new update, but I've decided I will make that initial invite (plea)
TL;DR
I hesitate to involve readers of radio-browser GitHub in our issue.

I joined [Notepad++ subreddit](https://www.reddit.com/r/notepadplusplus) and received a quick response, so manual edits of vandalized links was easy to find. (at least of the obvious kind), once the server passes QC, maybe it will be agreeable to finesse some of the data fields in a conversion. I need to learn more about the conversion process anyway.

Of course the more that has to be done the bigger the resource needs. Have you actually seen the number of genres in RadioSure using the 4-28 conversion? I'm surprised THAT doesn't crash the app. The genres people create get far too specific. 

Without any knowledge about programming language and the legality of abandonware, I found a decompler, but the result is gibberish to me. I don't know what to do with it.

**Comment by JazzfanRS** [comment]
I looked at the suggested sites, and none allow importing. as far as i can tell. And the english site still was in french and i had to wait for translation even to sign up a dummy account to try it out.

I am open to sharing the Google Sheet for a select few, I also am trying to figure out if GitHub has a spreadsheet function available, that would act like forks. If that is your meaning of 'traps'. If the file isn't protected and we allow someone to edit a Google Sheet we wouldn't be any better then radio-browser if we don't catch it. My estimate is 5 minutes to search radio-browser with as many tags as possible and then another 1-2  minutes entering the data. Thats an approximation on the high side hopefully.
Our conversion of their database compared to 'final.rsd' still freezes my PC. Is nearly 10 years old with only 8gb RAM for Windows 7 x64. When I ran a comparison of just the primary streams of both files I estimate only 10 or 20 matches. I didn't get to name searches  yet I can see there will be more matches there, so using RS to compare the feeds and the rotating banners will help. It's just a slow process regardless.

**Comment by JazzfanRS** [comment]
@GHbasicuser Did you not want access to upload to the repository? I just now noticed you had responded yet to the invite to collaberate.

**Comment by GHbasicuser** [comment]
The Framacalc tool is still fresh and I imagine that it's mainly used by a few users and associations in France. 😐 A more complete translation will surely come later. But its use is simple, rows and columns, copy and paste,... To manage a few tens (or hundreds) of names with comments to explain to others what/where to find, I believe it can be used by everyone.

On the other hand, I don't believe that there are only a few hundred radio stations in common between the two bases. 🤔 There must be dead links (on the radiosure side, because on the radiobrowser side they disappear within 7 days), and redirections in your link comparisons. :-| (eg: port number which differs from http to https, m3u8, pls,..)

Perhaps launching 2 instances of RadioSure at the same time (installed in portable mode, for example), one with the original base and the other with the RadioBrowser base, could help to better compare.. and possibly facilitate a copy-paste of the correct information to Radio-Browser.

I have already found most of the radio stations I listened to in the Radio-Browser list. It's also true that I had to add some to the base, it takes time to do that well... but that's something I also did with Radiosure frequently. And many others have done it too and that's how the base has gained thousands of links in a few years. 

I really believe that it's better that people (from all countries) come to Radio-Browser to put there the radios they know and which are not already there. In fact let them do as before with Radiosure.

I'm watching the evolution of this deposit and if I see the opportunity to upload I will. 😉 I am already satisfied to see that my idea of a converter (RB2RS) materializes. 😀

**Comment by francois-neosurf** [comment]
Hello !

Are you looking for a tool to compare txt files ?
Use : textpad. 
It does it automatically.

**Comment by GHbasicuser** [comment]
Don't ask me how (too long 😅), but from what i see there are at least 3300 radios in common (identical url). 🙂

And there are surely many (many) more because Radio-Browser include a lot of url with "playlist" (pls, m3u, m3u8,..) which should rather be avoided to propose to RadioSure (by preferring streaming direct access: mp3, aac). What surprises me is that RadioSure doesn't seem to have any problem reading these "pls, m3u,..". A good thing. 👍

(I also found over 1000 duplicates in Radio-Browser.)

**Comment by GHbasicuser** [comment]
In fact, I just noticed that in the original database, playlists (pls, m3u, etc) are avoided as much as possible. Maybe there's a good reason. 🤔

Example of what a pls (radio streaming) may contain:
```
[playlist]
numberofentries=5
File1=https://ice6.somafm.com/groovesalad-128-aac
Title1=SomaFM: Groove Salad (#1): A nicely chilled plate of ambient/downtempo beats and grooves.
Length1=-1
File2=https://ice1.somafm.com/groovesalad-128-aac
Title2=SomaFM: Groove Salad (#2): A nicely chilled plate of ambient/downtempo beats and grooves.
Length2=-1
File3=https://ice4.somafm.com/groovesalad-128-aac
Title3=SomaFM: Groove Salad (#3): A nicely chilled plate of ambient/downtempo beats and grooves.
Length3=-1
File4=https://ice2.somafm.com/groovesalad-128-aac
Title4=SomaFM: Groove Salad (#4): A nicely chilled plate of ambient/downtempo beats and grooves.
Length4=-1
File5=https://ice5.somafm.com/groovesalad-128-aac
Title5=SomaFM: Groove Salad (#5): A nicely chilled plate of ambient/downtempo beats and grooves.
Length5=-1
Version=2
```

Concerning Framacalc, I only see its use for dozens (or hundreds) of user requests (per month). If someone finds the radio, he adds it to Radio-Browser and writes something like that "Done (05/05/22 - by JazzfanRS)" 😅 in a cell (or removes the row) on the Framacalc page. (of course, there are surely many other spreadsheets online, but this one is free, without ads or click traps... ;-))

**Comment by JazzfanRS** [comment]
Yes, and I had to have it translate again as 'Previous Revisions' Menu item was in French. No idea how many revisions it will hold, Of course I don't care to go back more than a day or two if, for now only us two work at it. I think it is fine for now.

RadioSure kept a score of contributors, I'm just going to delete a line. I set up a second monitor, as it requires having a few pages open.  I have to watch the GPU temp as everything about my set up is slow speed and capacity for today's standards.

**Comment by JazzfanRS** [comment]
Damn. I think I managed to delete the spreadsheet. And return to 'last revision' didn't work reloaded. frama didn't copy lines 22626-29675. Maybe just load that into a separate page. In fact it might be good to separate it into 7000+ line files.

**Comment by JazzfanRS** [comment]
I did find one I like a little more than framacalc as its spreadsheet is a little bit more condensed so more lines are viewable, In fact it's very much like Google Sheets. I'll ask now, other than inability to rollback an edit was there any other aversion to using Google? I may not be able to rollback edits. But it keeps a log of activity. 

[Zoho](https://sheet.zoho.com/sheet/open/n86xcadbfa59abc5e49bbb229247270d0c877?sheet=stations-2022-03-10&range=G11) looks promising (tell me if you can see the spreadsheet and if you can alter it), but it appears I can add collaborators by email. And then you get invited to create an account to work on this file. 

They do connect you with the closest server, it detected I'm in Florida and set me to use the USA server. Sounds like a definite bonus. 

All in all, I decided we shouldn't sweat rollback too much. And just add them from our individual lists. I always do a quick check of name url and country anyway. Even give it a listen if to stations that match all three

At this point, I am going to filter the final update for US stations on Zoho and create a list just of them to work on myself. for now. And not deal with the others until I'm done  with my USA list.

**Comment by francois-neosurf** [comment]
Hi,
In my sql conversion, I can - for example - import only the stations which were not in last radiosure stations update.
I can also detect duplicates by making a comparison between the RB and the RS databases.
Please tell me what you need.

**Comment by JazzfanRS** [comment]
@francois-neosurf This is my exact reason for wanting to familiarize myself with MySQL, MariaDB, etc. 

Splendid! But after rewriting this 3 times, I admit I have reservations about the process of the conversions. 
I will make this a new Issue and provide my response there.
 
You are the sole coder but the content of fieldtypes (genres, etc) needs to be controllable. The genres are a mess because of the nature of Rb's submissions. I had hoped you could take the data from specific fields and apply a variable search and replace. `stations-2022-03-10.rsd` has 115 listed genre types, and of 29,674 stations only three are (-) or unknown. Your first 'official' update `stations-2022-05-06.rsd has 11,961 unique "genres". Nothing can be done to reduce the 7,406 self-named `unknown` type. It appears a third of the 30,000 +/- stations do not share the same 'genre' with another listing.
Here's a late thought: Most of the genres are a misinterpretation of the data of connected words separated by commas are seen as one fieldname. Can you take those items and just rename them as *multiple types* using a symbol to isolate then in sorting?
Variety is not a suitable term as it is already a broadcasting term used to define a station that combines the airing music, with talk-radio, etc.

You already take blank fields and output them as unknown. Though my preference is a (-) for blanks because symbols are sorted first whereas `unknown` places it within the alphabet. And visually a one character "fieldname" is easier to see. That would be a help for a someone wishing to submit an update to Rb with more precise and complete information.

**Comment by JazzfanRS** [comment]
I am supplying the unseen masses a listing that consists of JUST those stations that are missing for the time being.
 I split 3-10 into separate files. One will be just the ones missing (which should be researched beforehand before individually submitted) and one with station listings that appear as minor data changes. Those require individual analysis. RS is essential as rotating banner info will show data not revealed in a Detail check.
I want submissions to have the fields expected at Rb. As well as not duplicating stations. I don't want what we do here counterproductive to Rbs' goals, as I hope we might be able to influence stricter control over what is accepted there.

I have read the issues on their GitHub and forcing conformity to the submission process is non-existent, causing it to be barely controlled chaos. Albeit it does work, but the openness of data is a mess to navigate.

**Comment by JazzfanRS** [comment]
@francois-neosurf @GHbasicuser  Feel free to run a script extracting missing stations. The possible name changes and URL streams are so variable that you might want to include them as missing.

There is expectation that 'volunteers' do thorough work and search radio-browser to see if they are already there. I don't know how well their API is at finding duplicates, and when it deletes them. Submitting to Rb is a slow process, and I hadn't come up with a reasonable way to get our list into theirs. 

I'd been looking at form filling scripts and programs that can be set fill a form multiple times automatically. Unfortunately I am the last person to be able to do this. Suggestions?

I am splitting the missings, and putting the first (alpha sorted) 5000 on Google Sheets (less slow downs in processing), Anyone with link can edit. I get activity logs online, and notifications when 'Anonymous' makes a change.
I will find out in the next month of activity just how many volunteers will help. I will have a list offline and will continue submiting stations from the end of the missing list.

July or Aug? I will announce the remaining stations will be withdrawn and submitted via computer program. Whether that is true or not is irrelevant as I will continue to search and submit when the mood strikes.

Understand this, I don't have expectations you do anything at all in any certain length of time. I'm an easy going sort, and I don't like telling any what and when to do something. All of this is on your leisure. This project belongs to both of you.

I am figuring out what and how to transfer maintainer status to another still. I may need to keep it as I may not be able set up the discussion forum.

**Comment by JazzfanRS** [comment]
@francois-neosurf @GHbasicuser n’hésitez pas à exécuter un script d’extraction des stations manquantes. Les changements de nom et les flux d’URL possibles sont si variables que vous voudrez peut-être les inclure comme manquants.

Je sais qu’il existe des scripts ou des programmes qui peuvent automatiser le remplissage d’un formulaire plusieurs fois. Ce serait la meilleure solution. Si cela est possible, je commencerai à collecter les informations supplémentaires que RBI (radio-browser.info) attend.
Faites-moi savoir si c’est mieux pour vous de comprendre. J’ai supposé que ma plaisanterie se traduisait correctement.
